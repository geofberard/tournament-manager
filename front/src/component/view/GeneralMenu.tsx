import { Grid, SxProps, Theme, Typography } from "@mui/material";
import * as React from "react";
import { FC } from "react";
import { Image } from "../atom/Image";

const menuContainer: SxProps = {
    backgroundColor: "primary.dark",
    color: "primary.contrastText",
};

const logoContainer: SxProps = {
    display: "flex",
    alignItems: "center",
};

const titleContainer: SxProps = {
    display: "flex",
    width: "80%",
    justifyContent: "space-evenly",
    paddingBottom: { xs: "0", md: "0"},
    borderBottom: (theme: Theme) => `5px solid ${theme.palette.secondary.main}`,
};

const logo: SxProps = {
    width: { xs: "75px" },
};

export const GeneralMenu: FC = () => (
    <Grid
        container
        item
        direction={{ xs: "row"}}
        justifyContent={{ xs: "space-between"}}
        alignItems={{ xs: "stretch"}}
        sx={menuContainer}
    >
        <Grid item sx={logoContainer}>
            <Image src="img/scuf-logo.svg" alt="SCUF" sx={logo} />
        </Grid>
        <Grid sx={titleContainer} >
            <Typography variant="h3">Tous les matchs</Typography>
            <Typography variant="h3">Classement</Typography>
            <Typography variant="h3">Vos Matchs</Typography>
            <Typography variant="h3">Phases finales</Typography>
        </Grid>
        <Grid item />
    </Grid>
);