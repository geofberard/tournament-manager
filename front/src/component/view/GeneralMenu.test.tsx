import * as React from "react";
import { testSnapshot } from "../../test/TestUtils";
import { GeneralMenu } from "./GeneralMenu";

testSnapshot(<GeneralMenu />);
