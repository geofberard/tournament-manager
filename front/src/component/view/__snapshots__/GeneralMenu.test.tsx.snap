// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match last snapshot 1`] = `
<ForwardRef(Grid)
  container={true}
  direction="row"
  item={true}
  justifyContent="space-between"
  sx={
    Object {
      "backgroundColor": "primary.dark",
      "color": "primary.contrastText",
    }
  }
>
  <ForwardRef(Grid)
    item={true}
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <Image
      alt="SCUF"
      src="img/scuf-logo.svg"
      sx={
        Object {
          "width": "75px",
        }
      }
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    sx={
      Object {
        "alignItems": "center",
        "borderBottom": [Function],
        "display": "flex",
      }
    }
  >
    <ForwardRef(Typography)
      variant="h3"
    >
      Tous les matchs
    </ForwardRef(Typography)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    sx={
      Object {
        "alignItems": "center",
        "borderBottom": [Function],
        "display": "flex",
      }
    }
  >
    <ForwardRef(Typography)
      variant="h3"
    >
      Classement
    </ForwardRef(Typography)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    sx={
      Object {
        "alignItems": "center",
        "borderBottom": [Function],
        "display": "flex",
      }
    }
  >
    <ForwardRef(Typography)
      variant="h3"
    >
      Vos Matchs
    </ForwardRef(Typography)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    sx={
      Object {
        "alignItems": "center",
        "borderBottom": [Function],
        "display": "flex",
      }
    }
  >
    <ForwardRef(Typography)
      variant="h3"
    >
      Phases finales
    </ForwardRef(Typography)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
  />
</ForwardRef(Grid)>
`;
